forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/test/java/com/forgerock/openbanking/aspsp/as/api/authorisation/AuthorisationApiControllerTest.java

```diff
@@ -62,96 +70,126 @@
     private AMGatewayService amGatewayService;
     @Mock
     private DiscoveryConfig discoveryConfig;
-    @InjectMocks
-    private AuthorisationApiController authorisationApiController;
+    @Mock
+    private JwtOverridingService jwtOverridingService;
+    @Mock
+    private TokenUsageService tokenUsageService;
+
+    private AuthorisationApiController authorisationApiController ;
+
+    @Before
+    public void setUp() throws Exception{
+        this.authorisationApiController = new AuthorisationApiController(this.headLessAuthorisationService,
+                this.cryptoApiClient, this.tppStoreService, "cookieName", false, this.amGatewayService,
+                this.jwtOverridingService, this.tokenUsageService, this.discoveryConfig);
+    }
 
     @Test
-    public void shouldNotThrowExceptionWhenContainAllScopes() throws OBErrorException, OBErrorResponseException {
+    public void shouldNotThrowExceptionWhenContainAllScopes() throws OBErrorException, OBErrorResponseException,
+            InvalidTokenException, ParseException, IOException {
         // Given
+        String clientId = "98e119f6-196f-4296-98d4-f1a2f445bca2";
         List<String> responseTypes = List.of("code id_token");
         given(discoveryConfig.getSupportedResponseTypes()).willReturn(responseTypes);
         String jwt = toEncodedSignedTestJwt("jwt/authorisation.jwt");
         Tpp tpp = new Tpp();
         OIDCRegistrationResponse registrationResponse = new OIDCRegistrationResponse();
-        registrationResponse.setJwks_uri("");
+        registrationResponse.setJwks_uri("url");
         tpp.setRegistrationResponse(registrationResponse);
-        given(tppStoreService.findByClientId(null)).willReturn(Optional.of(tpp));
+        given(tppStoreService.findByClientId(clientId)).willReturn(Optional.of(tpp));
+        SignedJWT signedJwt = mock(SignedJWT.class);
+        given(cryptoApiClient.validateJws(anyString(), anyString(), anyString() )).willReturn(signedJwt);
 
         // When
-        authorisationApiController.getAuthorisation(responseTypes.get(0), null, null, null, "openid accounts payments", null, jwt, true, null, null, null, null, null);
+        authorisationApiController.getAuthorisation(responseTypes.get(0), clientId, null,
+                null, "openid accounts payments", null, jwt, true, null, null, null, null, null);
 
         // Then no exception
```

Comment:
Or you could assert something in the response?

Reply:
Fixed